### PR TITLE
Cache shots in Dribbble repo and retrieve in the VM.

### LINKED ISSUE #383: [DR] Cache shots & retrieve by ID
- [ ] Add an in-memory cache of shots to repository
- [ ] Update `DribbbleShot` to retrieve shot by ID
- [ ] Remove parceling of `Shot`

### PR DESCRIPTION
**Pull request by nickbutcher**
Fixes #383

### CODE REVIEW

File: dribbble/src/main/java/io/plaidapp/dribbble/ui/shot/DribbbleShotViewModel.kt [hunk 1]
```diff
@@ -31,11 +33,23 @@ import kotlinx.coroutines.experimental.launch
...
 
-    var shot: Shot? = null // TODO retrieve this from the repo
+    val shot: Shot
+
+    init {
+        val result = shotRepository.getShot(shotId)
+        if (result is Result.Success) {
+            shot = result.data
+        } else {
+            throw IllegalStateException("Could not retrieve shot $shotId")
```
Comment (florina-muntenescu) [inline dribbble/src/main/java/io/plaidapp/dribbble/ui/shot/DribbbleShotViewModel.kt]: How about throwing the exception from the Result?

File: core/src/main/java/io/plaidapp/core/dribbble/data/DribbbleRepository.kt [hunk 2]
```diff
@@ -42,6 +44,15 @@ class DribbbleRepository(
         inflight[id] = launchSearch(query, page, id, onResult)
     }
 
+    fun getShot(id: Long): Result<Shot> {
```
Comment (florina-muntenescu) [inline core/src/main/java/io/plaidapp/core/dribbble/data/DribbbleRepository.kt]: all of the new logic from DribbbleRepository is not tested

File: dribbble/src/main/java/io/plaidapp/dribbble/ui/shot/DribbbleShotViewModel.kt (hunk 1 above)
Comment (nickbutcher) [reply dribbble/src/main/java/io/plaidapp/dribbble/ui/shot/DribbbleShotViewModel.kt]: As discussed leaving for now until we decided about `Result.Loading` case.

File: core/src/main/java/io/plaidapp/core/dribbble/data/DribbbleRepository.kt (hunk 2 above)
Comment (nickbutcher) [reply core/src/main/java/io/plaidapp/core/dribbble/data/DribbbleRepository.kt]: done.

**Pull request merged**